### DISCUSSION TITLE
[Android] How can I create/open new Activity

### DISCUSSION THREAD

**Discussion by furesoft** [AvaloniaUI/Avalonia#9825] [Q&A]
Hi,

I want to open a settingsview after button click. How could I achieve this?

**Comment by timunie** [comment]
Hi @furesoft ,

Normally I'd use the MVVM approach. So I'd have a ViewModel for Settings and a MainViewModel. Inside a Command I would set Settings as the new Content. DataTemplates will do the rest. 

See https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src%2FAvalonia.Samples%2FRouting%2FBasicViewLocatorSample

**Reply by furesoft** [reply]
Yeah, but then the Android Back Button does not work?

**Reply by timunie** [reply]
I think @emmauss is working on a control for it. May be available in the all nuget feed to try out, but I'm not sure. https://github.com/AvaloniaUI/Avalonia/pull/9352

**Reply by maxkatz6** [reply]
> Yeah, but then the Android Back Button does not work?

You can handle it using android API.

**Reply by maxkatz6** [reply]
You also can create native android activities and put AvaloniaView inside, if you prefer. Use AvaloniaMainActivity as a reference.

**Reply by AArnott** [reply]
The sample that @timunie linked to is helpful, but it doesn't describe how to hook into Android's Back button. The only access I can find to the back button is in the `AvaloniaMainActivity`-derived class. But I can't find the view or view model from this class, so I don't know how to pass this back navigation event to the view model so that it can change its CurrentPage property. Any suggestions?

**Reply by maxkatz6** [reply]
Abstracted API, which also works on the Browser backend, is TopLevel.BackRequested event. You don't need access to AvaloniaMainActivity for that. Just `TopLevel.GetTopLevel(view).BackRequested`.
